index.js

```diff
@@ -573,8 +574,23 @@ class Raspi extends EventEmitter {
     throw new Error('reportDigitalPin is not yet implemented');
   }
 
-  pingRead() {
-    throw new Error('pingRead is not yet implemented');
+  pingRead(config, handler) {
+    const pulseOut = config.pulseOut;
+    const pin = config.pin;
+    const sonar = new Sonar(config.pin);
+    
+    const pinInstance = this[getPinInstance](this.normalize(pin));
+    if (pinInstance.mode != INPUT_MODE) {
+      this.pinMode(pin, INPUT_MODE);
```

Comment:
I think that's fine, since Johnny-Five won't be setting a mode, you can use whatever mechanism you need to track it internally.